File: AtlasModel/Enterprise/TTFormFieldDef.cs
using System;

namespace AtlasModel.Enterprise
{
    public partial class TTFormFieldDef
    {
        public Guid FieldDefId { get; set; }
        public Guid FormDefId { get; set; }
        public Guid? ParentFieldDefId { get; set; }
        public int OrderNo { get; set; }
        public string Name { get; set; }
        public string ControlClass { get; set; }
        public string DataMember { get; set; }
        public string ControlProperties { get; set; }
        public short CheckOutStatus { get; set; }
    }
}

File: AtlasWeb/Core/wwwroot/app/NebulaClient/StorageManager/DefinitionManagement/TTFormFieldDef.ts
import { Guid } from "../../Mscorlib/Guid";
import { TTDefCollection } from "./TTDefCollection";
import { TTFormFieldEventDef } from "./TTFormFieldEventDef";
import { TTUnboundFormDef } from "./TTUnboundFormDef";

export class TTFormFieldDef {
    public FieldDefID: Guid;
    public FormDef: TTUnboundFormDef;
    public FormDefID: Guid;
    public ParentFieldDef: TTFormFieldDef;
    public ParentFieldDefID: Guid;
    public Name: string;
    public OrderNo: number;
    public ControlClass: string;
    public ControlInterface: string;
    public DataMember: string;
    public ControlProperties: string;
    public EventDefs: TTDefCollection<TTFormFieldEventDef>;
    public IsGrid: boolean;
    public IsGridColumn: boolean;
    public ToString(): string {
        return null;
    }
}
